File: notes_app/lib/services/auth_service.dart
import 'package:appwrite/appwrite.dart';
import 'package:appwrite/models.dart';
import '../config/appwrite_config.dart';

class AuthService {
  late final Account account;

  AuthService() {
    account = Account(AppwriteConfig.client);
  }

  Future<Session> createAccount(String email, String password, String name) async {
    try {
      final user = await account.create(
        userId: ID.unique(), email: email, password: password, name: name,
      );
      if (user.$id.isNotEmpty) {
        return login(email, password);
      } else {
        throw Exception('Failed to create account');
      }
    } catch (error) {
      rethrow;
    }
  }

  Future<Session> login(String email, String password) async {
    try {
      return await account.createEmailPasswordSession(email: email, password: password);
    } catch (error) {
      rethrow;
    }
  }

  Future<User?> getCurrentUser() async {
    try {
      return await account.get();
    } catch (error) {
      return null;
    }
  }

  Future<void> logout() async {
    try {
      await account.deleteSession(sessionId: 'current');
    } catch (error) {
      rethrow;
    }
  }
}

File: NotesApp/src/services/auth.service.js
import { ID } from "appwrite";
import { appwriteClient } from "../config/appwrite";

class AuthService {
  account;

  constructor() {
    this.account = appwriteClient.account;
  }

  async createAccount(email, password, name) {
    try {
      const userAccount = await this.account.create(ID.unique(), email, password, name);
      if (userAccount) {
        return this.login(email, password);
      } else {
        return userAccount;
      }
    } catch (error) {
      console.error("Error creating account:", error);
      throw error;
    }
  }

  async login(email, password) {
    try {
      return await this.account.createEmailSession(email, password);
    } catch (error) {
      console.error("Error logging in:", error);
      throw error;
    }
  }

  async getCurrentUser() {
    try {
      return await this.account.get();
    } catch (error) {
      console.error("Error getting current user:", error);
      return null;
    }
  }

  async logout() {
    try {
      return await this.account.deleteSession("current");
    } catch (error) {
      console.error("Error logging out:", error);
      throw error;
    }
  }
}

const authService = new AuthService();
export default authService;
